src/components/ChartArea/ChartPicker.jsx

```diff
@@ -0,0 +1,98 @@
+import {Form, FORM_LAYOUTS} from "../Form/Form";
+import {FormSelectorField} from "../Form/Fields/FormFields";
+import {CHART_AREA_FIELDS} from "../../common/fields";
+import React, {useEffect, useState} from "react";
+import {isSet} from "../Form/Validation/formValidations";
+import {CHART_AREA_FORM_INITIAL_VALUES} from "../../common/constants";
+import {ChartPickerContainer, TitleContainer} from "./Styles";
+import {Title} from "../ChemicalModels/Models/ModelsStyles";
+import {getCurves} from "../../services/processes";
+import {MessageContainer} from "../IdealAdsorbentList/Styles";
+import Typography from "@material-ui/core/Typography";
+
+export const ChartPicker = ({onSubmit, processId}) => {
+  const [errorValues, setErrorValues] = useState({});
+  const [charts, setCharts] = useState(null);
+  const [baseLines, setBaseLines] = useState(null);
+  useEffect(async () => {
+    const response = await getCurves(processId);
+
+    setCharts(
+      response
+        .filter((chart) => !chart.esLineaBase)
+        .map((chart) => {
+          return {
+            label: chart.nombre,
+            value: chart.id,
+          };
+        }),
+    );
+    setBaseLines(
+      response
+        .filter((chart) => chart.esLineaBase)
+        .map((chart) => {
+          return {
+            label: chart.nombre,
+            value: chart.id,
+          };
+        }),
+    );
+  }, [processId]);
```

Comment:
Este segundo map es igual al anterior y me parece que podriamos evitarlo haciendo todo en el mismo. Por un lado podriamos filtrar y guardar en un array los que son linea base y en otro los que no y nos ahorramos un .map

Reply:
How about this
```javascript
  useEffect(async () => {
    const response = await getCurves(processId);
    const [charts, baseLines] = [[], []];

    response
      .map((curve) => {
        return {
          label: curve.nombre,
          value: curve.id,
        };
      })
      .forEach((curve) => {
        curve.esLineaBase ? baseLines.push(curve) : charts.push(curve);
      });

    setCharts(charts);
    setBaseLines(baseLines);
  }, [processId]);
```